### PR TITLE
Basic Obstacle avoidance

### LINKED ISSUE #6: 1.2 Basic obstacle avoidance
- As an employee I want the assistant to be able to avoid obstacles so that it doesn’t drive into objects or customers. 

**Complete criteria: ** 
When the vehicle faces an obstacle it should stop and stay stationary until the object has been moved.  

**Link to Wiki: ** https://github.com/DIT113-V22/group-15/wiki/Milestone-1-Requirements

### PR DESCRIPTION
**Pull request by ghost**
This will enable the vehicle to avoid obstacle in different states.
**1**. If there is an obstacle at the front and behind. 
**2.** As well as if there is obstacle front and on one side of the vehicle it will turn to the direction where there are no obstacle.
**3.** If there is an obstacle front, left and right it will reverse. 

**Comments:** 
In case 3 is difficult to test in the current environment but we believe in theory it will work.

@AlaahT 
@JonnaJ98 
@anselin-thefirst 
@LabibaKararEshaba 
@Lexzet21

### CODE REVIEW

**Review by platisd — CHANGES_REQUESTED** [review]
The code looks fine but it's not built on CI

**Comment by platisd** [comment]
A reminder: This cannot be approved until you address the comment.

**Comment by Dahlaan** [comment]
Hi. We are a little bit confused here, it passed the Github action with no problem what are we missing? Is Github actions not the CI? Or are we missing workflows? 
Best Regards 
Carl DT

**Comment by platisd** [comment]
The workflow is not building the new sketch you have committed but just the old one.

**Comment by platisd** [comment]
Well this fixes things indeed, but do you understand why the problem existed in the first place? It's not because "you must make changes to the smartcar.ino file that already exists".

**Review by platisd — APPROVED** [review]

**Comment by Dahlaan** [comment]
I think so? Is it because the file name was different from from smartcar.ino? As long the name matches the job will run?

**Comment by platisd** [comment]
> I think so? Is it because the file name was different from from smartcar.ino? As long the name matches the job will run?

Or you can simply change the workflow to reflect your project structure and filenames.

**Comment by Dahlaan** [comment]
Okay then we understand! Thank you!

File: sketch_obstacle_aavoidance.ino [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include <Smartcar.h>
```
Comment (platisd) [inline sketch_obstacle_aavoidance.ino:1]: This shouldn't have been merged. I must have missed this or it showed up in the meanwhile.

Comment (Dahlaan) [reply sketch_obstacle_aavoidance.ino:1]: Sorry i don't understand, can you elaborate? @platisd

Comment (platisd) [reply sketch_obstacle_aavoidance.ino:1]: What is the point of this file? It looks like a copy of the `arduino/smartcar/smartcar.ino`

Comment (Dahlaan) [reply sketch_obstacle_aavoidance.ino:1]: Oh i didn't see that there was two files, i don't know what happened here. I don't know why there is copy of this file. i will remove the "sketch_obstacle_aavoidance.ino" sketch.

**Pull request merged**